core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/AbstractFileBatchSource.java

```diff
@@ -96,7 +96,11 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     if (!config.containsMacro("timeTable") && config.timeTable != null) {
       pipelineConfigurer.createDataset(config.timeTable, KeyValueTable.class, DatasetProperties.EMPTY);
     }
-    pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+    if (config.getSchema() != null) {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+    } else {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(DEFAULT_SCHEMA);
```

Comment:
This isn't correct. For example, if schema is null and format is avro, you will set the output schema to the default schema, which is almost certainly wrong.

You should be setting the actual output schema here. The same schema you set in the input format is what should be set here.